### DISCUSSION TITLE
V5 版本 changeData 后会丢失 legend 激活状态，V4 版本中会维持 legend 激活状态

### DISCUSSION THREAD

**Discussion by pfdgithub** [antvis/G2#5611] [Q&A]
- **G2 Version**: 5.1.5
- **Platform**: macOS Chrome 117.0.5938.149 (arm64)
- **Mini Showcase(like screenshots)**:
- **CodePen Link**:

使用官网 Demo 即可验证。
V4 版本 Demo 中在切换 legend 激活状态后 changeData 能继续保持更新前的激活状态。
V5 版本 Demo 中，不会保留激活状态，这是设计如此？还是重构后功能缺失？或者就是个 BUG ？

**Comment by pearmini** [comment]
目前的设计就是如此，解决办法是：

- 通过 `chart.on('legend:filter', callback)` 去记录当前的状态
- 每次 changeData 之后通过 `chart.emit('legend:filter')` 去恢复状态

**Reply by pfdgithub** [reply]
之前就是通过 `chart.emit('legend:filter')` 去恢复的，但似乎仍会在某些情况下丢失激活状态。似乎是在更新图表配置事出现的，目前这个需求被砍了，就没又细查。
